[CI] FileSettingsRoleMappingsRestartIT testReservedStatePersistsOnRestart failing

**Build Scans:**
- [elasticsearch-intake #16610 / part4](https://gradle-enterprise.elastic.co/s/j2v4pu76jfxsg)
- [elasticsearch-periodic-platform-support #6045 / almalinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/oourssst7yjog)
- [elasticsearch-periodic-platform-support #6045 / ubuntu-2204_platform-support-unix](https://gradle-enterprise.elastic.co/s/6v3acngjdwcco)
- [elasticsearch-pull-request #53316 / part-4](https://gradle-enterprise.elastic.co/s/guz5z6xcrl3uo)
- [elasticsearch-pull-request #53266 / part-4](https://gradle-enterprise.elastic.co/s/zoq2atxook44k)
- [elasticsearch-periodic-platform-support #6040 / rhel-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/ofycwc77ddygo)
- [elasticsearch-periodic-platform-support #6040 / almalinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/3aou7trys3ptg)
- [elasticsearch-periodic-platform-support #6040 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/taufhivzocx2e)
- [elasticsearch-periodic #6051 / openjdk23_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/3imxp3jr4otdy)
- [elasticsearch-periodic #6046 / openjdk23_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/b6ehykxo62edk)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:security:internalClusterTest" --tests "org.elasticsearch.xpack.security.FileSettingsRoleMappingsRestartIT.testReservedStatePersistsOnRestart" -Dtests.seed=84B6A2BE7D7481F2 -Dtests.locale=sr -Dtests.timezone=America/Los_Angeles -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.security.FileSettingsRoleMappingsRestartIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testReservedStatePersistsOnRestart),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: iterable with items [] in any order
     but: no match for: "everyone_kibana_alone"
```

**Issue Reasons:**
- [8.x] 2 consecutive failures in step almalinux-9_platform-support-unix
- [8.x] 11 failures in test testReservedStatePersistsOnRestart (6.5% fail rate in 168 executions)
- [8.x] 3 failures in step part-4 (7.5% fail rate in 40 executions)
- [8.x] 2 failures in step almalinux-9_platform-support-unix (66.7% fail rate in 3 executions)
- [8.x] 2 failures in step openjdk23_checkpart4_java-matrix (66.7% fail rate in 3 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic (66.7% fail rate in 3 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic-platform-support (66.7% fail rate in 3 executions)
- [8.x] 3 failures in pipeline elasticsearch-pull-request (7.9% fail rate in 38 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -158,9 +158,6 @@ tests:
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test140CgroupOsStatsAreAvailable
   issue: https://github.com/elastic/elasticsearch/issues/120914
-- class: org.elasticsearch.xpack.security.FileSettingsRoleMappingsRestartIT
-  method: testReservedStatePersistsOnRestart
-  issue: https://github.com/elastic/elasticsearch/issues/120923
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test070BindMountCustomPathConfAndJvmOptions
   issue: https://github.com/elastic/elasticsearch/issues/120910
@@ -173,9 +170,6 @@ tests:
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=nodes.stats/11_indices_metrics/indices mappings exact count test for indices level}
   issue: https://github.com/elastic/elasticsearch/issues/120950
-- class: org.elasticsearch.xpack.security.FileSettingsRoleMappingsRestartIT
-  method: testFileSettingsReprocessedOnRestartWithoutVersionChange
-  issue: https://github.com/elastic/elasticsearch/issues/120964
 - class: org.elasticsearch.xpack.ml.integration.PyTorchModelIT
   issue: https://github.com/elastic/elasticsearch/issues/121165
 - class: org.elasticsearch.xpack.security.authc.ldap.ActiveDirectorySessionFactoryTests

--- a/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/FileSettingsRoleMappingsRestartIT.java
+++ b/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/FileSettingsRoleMappingsRestartIT.java
@@ -114,7 +114,7 @@ public class FileSettingsRoleMappingsRestartIT extends SecurityIntegTestCase {
         internalCluster().setBootstrapMasterNodeIndex(0);
 
         final String masterNode = internalCluster().startMasterOnlyNode();
-        awaitMasterNode(masterNode, masterNode);
+        awaitMasterNode();
         var savedClusterState = setupClusterStateListener(masterNode, "everyone_kibana_alone");
 
         awaitFileSettingsWatcher();
@@ -195,7 +195,7 @@ public class FileSettingsRoleMappingsRestartIT extends SecurityIntegTestCase {
         internalCluster().setBootstrapMasterNodeIndex(0);
 
         final String masterNode = internalCluster().startMasterOnlyNode();
-        awaitMasterNode(masterNode, masterNode);
+        awaitMasterNode();
 
         Tuple<CountDownLatch, AtomicLong> savedClusterState = setupClusterStateListener(masterNode, "everyone_kibana_alone");
         awaitFileSettingsWatcher();

--- a/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/FileSettingsRoleMappingsRestartIT.java
+++ b/x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/FileSettingsRoleMappingsRestartIT.java
@@ -113,7 +113,8 @@ public class FileSettingsRoleMappingsRestartIT extends SecurityIntegTestCase {
     public void testReservedStatePersistsOnRestart() throws Exception {
         internalCluster().setBootstrapMasterNodeIndex(0);
 
-        final String masterNode = internalCluster().getMasterName();
+        final String masterNode = internalCluster().startMasterOnlyNode();
+        awaitMasterNode(masterNode, masterNode);
         var savedClusterState = setupClusterStateListener(masterNode, "everyone_kibana_alone");
 
         awaitFileSettingsWatcher();
@@ -193,7 +194,8 @@ public class FileSettingsRoleMappingsRestartIT extends SecurityIntegTestCase {
     public void testFileSettingsReprocessedOnRestartWithoutVersionChange() throws Exception {
         internalCluster().setBootstrapMasterNodeIndex(0);
 
-        final String masterNode = internalCluster().getMasterName();
+        final String masterNode = internalCluster().startMasterOnlyNode();
+        awaitMasterNode(masterNode, masterNode);
 
         Tuple<CountDownLatch, AtomicLong> savedClusterState = setupClusterStateListener(masterNode, "everyone_kibana_alone");
         awaitFileSettingsWatcher();

